types/stompjs/index.d.ts

```diff
@@ -26,7 +26,7 @@ export class Client {
 
     debug(...args: string[]): any;
 
-    connect(headers: { login: string, passcode: string, host?: string }, connectCallback: (frame?: Frame) => any, errorCallback?: (error: string) => any): any;
+    connect(headers: { login?: string, passcode?: string, host?: string }, connectCallback: (frame?: Frame) => any, errorCallback?: (error: string) => any): any;
```

Comment:
According to https://github.com/jmesnil/stomp-websocket/blob/master/src/stomp.coffee#L215 there is a good reason they should always be present when `headers` is passed. I haven't found a case yet where `login` and `passcode` isn't required.

Reply:
I certainly have. I don't see a good reason that they are always present in `stomp.coffee` that you reference. Can you please clarify?

Reply:
From my experience, a connection to a server should always be authenticated. Even if credentials are as simple as `quest` like in `RabbitMQ`.

Reply:
Now, in the case where credentials are not necessary (at your own risk) `client.connect({ login: null, passcode: null }, () => { })` shoudl suffice.

Reply:
Indeed, but there are certainly cases where authentication could take place separately. I don't think a type library is the place to go about changing the protocol definition, do you?

Reply:
The protocol itself says the headers are optional. The library allows for you to omit them and doesn't throw if you do. Nowhere is it written that they are mandatory for all possible STOMP connections or all users of this library. They seem perfectly optional to me.

Reply:
Indeed looking at the code again they are optional as the author assigned an empty object if they are not passed.
In this case, I think it's best to add more overrides just to make the use cases clearer.

```
connect(connectCallback: (frame?: Frame) => any, errorCallback?: (error: string) => any): any;
```